src/main/java/goodgenerator/blocks/tileEntity/ComponentAssemblyLine.java

```diff
@@ -0,0 +1,744 @@
+                    + EnumChatFormatting.RESET + " EU/t(*" + EnumChatFormatting.YELLOW
+                    + GT_Utility.formatNumbers(
+                            GT_ExoticEnergyInputHelper.getMaxInputAmpsMulti(getExoticAndNormalEnergyHatchList()))
+                    + EnumChatFormatting.RESET + "A) " + StatCollector.translateToLocal("GT5U.machines.tier")
+                    + ": " + EnumChatFormatting.YELLOW
+                    + VN[
+                            GT_Utility.getTier(GT_ExoticEnergyInputHelper.getMaxInputVoltageMulti(
+                                    getExoticAndNormalEnergyHatchList()))]
+                    + EnumChatFormatting.RESET,
+            StatCollector.translateToLocal("scanner.info.CASS.tier")
+                    + (casingTier >= 0 ? GT_Values.VN[casingTier + 1] : "None!")
+        };
+    }
+
+    @Override
+    public ITexture[] getTexture(
+            IGregTechTileEntity aBaseMetaTileEntity,
+            byte aSide,
+            byte aFacing,
+            byte aColorIndex,
+            boolean aActive,
+            boolean aRedstone) {
+        if (aSide == aFacing) {
+            if (aActive)
+                return new ITexture[] {
+                    Textures.BlockIcons.getCasingTextureForId(183),
+                    TextureFactory.builder()
+                            .addIcon(OVERLAY_FRONT_PROCESSING_ARRAY_ACTIVE)
+                            .extFacing()
+                            .build(),
+                    TextureFactory.builder()
+                            .addIcon(OVERLAY_FRONT_PROCESSING_ARRAY_ACTIVE_GLOW)
+                            .extFacing()
+                            .glow()
+                            .build()
+                };
+            return new ITexture[] {
+                Textures.BlockIcons.getCasingTextureForId(183),
+                TextureFactory.builder()
+                        .addIcon(OVERLAY_FRONT_PROCESSING_ARRAY)
+                        .extFacing()
+                        .build(),
+                TextureFactory.builder()
+                        .addIcon(OVERLAY_FRONT_PROCESSING_ARRAY_GLOW)
+                        .extFacing()
+                        .glow()
+                        .build()
+            };
+        }
+        return new ITexture[] {Textures.BlockIcons.getCasingTextureForId(183)};
+    }
+
+    @Override
+    public boolean isCorrectMachinePart(ItemStack aStack) {
+        return true;
+    }
+
+    @Override
+    public boolean checkRecipe(ItemStack aStack) {
+        long tVoltage = GT_ExoticEnergyInputHelper.getMaxInputVoltageMulti(getExoticAndNormalEnergyHatchList());
+        long tAmps = GT_ExoticEnergyInputHelper.getMaxInputAmpsMulti(getExoticAndNormalEnergyHatchList());
+        long totalEU = tVoltage * tAmps;
+        ItemStack[] tItems = getCompactedInputs();
+        FluidStack[] tFluids = getCompactedFluids();
```

Comment:
use `getStoredFluids()` and `getStoredInputs()`, otherwise it can't recongize items that over one stacks

Reply:
fixed in [4d8ec19](https://github.com/GTNewHorizons/GoodGenerator/pull/102/commits/4d8ec19ba2e385e784c500db43f4ae98a089a445)